src/collectionview/shared-pool.android.ts

```diff
@@ -0,0 +1,96 @@
+import { SharedCollectionViewPoolBase } from './shared-pool-common';
+
+interface CollectionViewCellHolder extends com.nativescript.collectionview.CollectionViewCellHolder {
+    // tslint:disable-next-line:no-misused-new
+    new (androidView: android.view.View): CollectionViewCellHolder;
+    view: View;
+    clickListener: android.view.View.OnClickListener;
+}
+
+export class SharedCollectionViewPool extends SharedCollectionViewPoolBase {
+    private recyclerListener: androidx.recyclerview.widget.RecyclerView.RecyclerListener;
+    private recycledViewPool: com.nativescript.collectionview.RecycledViewPool;
+    private recycledViewPoolDisposeListener: com.nativescript.collectionview.RecycledViewPool.ViewPoolListener;
+
+    // used to store viewHolder and thus their corresponding Views
+    // used to "destroy" cells when possible
+    public _viewHolders = new Set<CollectionViewCellHolder>();
+
+    public createNativeView() {
+        // no need to create anything
+        return null;
+    }
+
+    public initNativeView(): void {
+        // const nativeView = this.nativeViewProtected;
+        this.recycledViewPool = new com.nativescript.collectionview.RecycledViewPool();
+        this.recycledViewPoolDisposeListener = new com.nativescript.collectionview.RecycledViewPool.ViewPoolListener({
+            onViewHolderDisposed: (holder: CollectionViewCellHolder) => {
+                if (Trace.isEnabled()) {
+                    CLog(CLogTypes.log, 'onViewHolderDisposed', holder);
+                }
+                if (this._viewHolders) {
+                    this._viewHolders.delete(holder);
+                }
+                const isNonSync = holder['defaultItemView'] === true;
+                const view = isNonSync ? (holder.view as ContentView).content : holder.view;
+                this.notifyForItemAtIndex(CollectionViewBase.itemDisposingEvent, view, holder.getAdapterPosition(), view.bindingContext, holder);
+                if (view && view.isLoaded) {
+                    view.callUnloaded();
+                }
+                view._isAddedToNativeVisualTree = false;
+                //@ts-ignore
+                view.parent = null;
+                view._tearDownUI();
+            }
+        });
+        (this.recycledViewPool as any).mListener = this.recycledViewPoolDisposeListener;
+
+        this.recyclerListener = new androidx.recyclerview.widget.RecyclerView.RecyclerListener({
+            onViewRecycled: (holder: CollectionViewCellHolder) => {
+                if (Trace.isEnabled()) {
+                    CLog(CLogTypes.log, 'onViewRecycled', holder);
+                }
+                const isNonSync = holder['defaultItemView'] === true;
+                const view = isNonSync ? (holder.view as ContentView).content : holder.view;
+                this.notifyForItemAtIndex(CollectionViewBase.itemRecyclingEvent, view, holder.getAdapterPosition(), view.bindingContext, holder);
```

Comment:
This should also likely be fired for all CollectionView instances as well (perhaps `this.notifyForItemAtIndex` would just notify self + all attached collectionView instances.